packages/esm-appointments-app/src/form/appointments-form.component.tsx

```diff
@@ -51,8 +50,11 @@ import {
 } from '../constants';
 import styles from './appointments-form.scss';
 import SelectedDateContext from '../hooks/selectedDateContext';
-import uniqBy from 'lodash-es/uniqBy';
-import { useController, Control } from 'react-hook-form';
+
+const time12HourFormatRegexPattern = '(1[0-2]|0?[1-9]):[0-5][0-9]';
+function isValidTime(timeStr) {
+  return timeStr.match(new RegExp(time12HourFormatRegexPattern));
+}
```

Comment:
Simple validation function to check the time format - used within the carbon field component to control whether the "invalid time" message appears, and within the form schema to control whether the form submission is allowed.